packages/ui/src/components/Button.tsx

```diff
@@ -37,14 +45,29 @@ export const Button: FC<Props> = ({
           ? "hover:bg-green-800 hover:border-green-800 hover:text-white"
           : "hover:bg-slate-200 hover:text-gray-900",
         wide && "w-full",
-        size === "medium" && "px-4 h-8 rounded-md",
-        size === "small" && "px-3 h-6 rounded-sm"
+        size === "medium" && "h-8 rounded-md",
+        size === "small" && "h-6 rounded-sm",
+        "[.button-group_&:not(:first-child)]:rounded-l-none",
+        "[.button-group_&:not(:first-child)]:border-l-0",
+        "[.button-group_&:not(:last-child)]:rounded-r-none"
```

Comment:
Sorry :) This probably could be simplified, but I spend ~30m fighting Tailwind because for some reason it's "[styling based on parent state](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state)" is based on parent group _state_, not on parent existing.

I was surprised you can't do `<div className="group/foo"><div className="group-foo:border">...</div></div>`.

Reply:
I suggest just adding a comment saying something to this extent. Next time someone comes across it, if there is a next time, they can choose to try to refactor.